test/umzug.test.ts

```diff
@@ -91,6 +91,49 @@ describe('custom context', () => {
 
 		expect(spy).toHaveBeenCalledTimes(1);
 	});
+
+	describe(`resolve asynchronous context getter before the migrations run`, () => {
+		const sleep = async (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+		const getContext = async () => {
+			// It guarantees the initialization scripts or asynchronous stuff finished their work
+			// before the actual migrations workflow begins.
+			// Eg: const externalData = await retrieveExternalData();
+			await sleep(100);
+
+			return {
+				// Eg: externalData,
+				innerValue: 'text',
+			};
```

Comment:
nit: this comment is confusing/not necessary - the one above explaining the use case you're trying to replicate is, but no need for the extra comment here:
```suggestion
			return { innerValue: 'text' };
```